File: STANDARD TEMPLATES/3elements.cpp
/***********************************************************************/
//  																		
//	INFINITUM 					                                    	
//	Rakesh Mahadasa														
//	National Institute of Technology , Calicut						
//	problem link : 
//	Concept : 
//	Reference :	
//																		
/***********************************************************************/


//header files

#include <iostream>
#include <string.h>
#include <algorithm>
#include <vector>
#include <stack>
#include <queue>
#include <cstdio>
#include <iomanip>
#include <cmath>
#include <set>
#include <map>
#include <ctime>
#include <limits.h>

//typedefs

typedef unsigned long long int ull;
typedef long long int ll;
typedef long int li;

#define MOD7 1000000007
#define MOD9 1000000009
#define PI 3.14159265


using namespace std;

ll gcd(ll a,ll b){
	while(b){
		ll temp = b;
		b=a%b;
		a=temp;
	}
	return a;
}


struct node{
	node* left;
	node* right;
	int val;
};
int low,high;
node* bst(node* root,int n){
	if(root==NULL){
		root->left = NULL;
		root->right = NULL;
		root->val=n;
		return root;
	}
	else if (root->val > n){
		high++;
		j
	}

}

int main(){
	int t;
	scanf("%d",&t);
	while(t--){
		
	}
	return 0;
}

File: Codechef/CodeVaitam - NIT A/3elements.py
# INFINITUM
# Rakesh Mahadasa
# National Institute of Technology, Calicut
# Problem Link : http://www.codechef.com/CDVA2015/problems/CDVA1510
# Concept : Simple Math
# Reference : None

n=input()
a = [int(x) for x in raw_input().split()]
curr_max = a[0];
b={a[0]:1}
el=[a[0]]
for x in a[1:]:
	if(x<=curr_max):
		if x == curr_max:
			b[x]+=1
		else:
			b[x]=1
			el.append(x)
			curr_max = x

l = len(el)
res = 0		
add = [b[el[0]]]
for x in range(1,l):
	add.append(add[x-1]+b[el[x]])


resl = add[0]
for x in range(1,l-1):
	resl = add[x-1]
	resr = add[l-1]-add[x]
	resn = add[x]-add[x-1]
	res+=resl*resr*resn
print(res)			
